test_runner/src/main/kotlin/ftl/json/SavedMatrix.kt

```diff
@@ -104,7 +109,9 @@ private fun SavedMatrix.updateOutcome(outcome: List<TestOutcome>) = copy(
     testAxises = outcome
 )
 
-private fun invalidTestOutcome() = TestOutcome(
-    outcome = "---",
-    details = "Matrix is invalid"
+private fun TestMatrix.invalidTestOutcome() = TestOutcome(
+    outcome = INVALID,
+    details = invalidMatrixDetails.orUnknown()
 )
+
+fun SavedMatrix.errorMessage() = "Matrix: [$matrixId] failed: ".plus(getOutcomeMessageByKey(testAxises.firstOrNull()?.details.orUnknown()))
```

Comment:
I think this extension function is used only in `MatrixMap`. How about moving it there and make it private?
(yup, we would need to update the unit test but I think we can have an error message hardcoded there)

Reply:
Right, thanks!